src/api/controllers/user.spec.ts

```diff
@@ -0,0 +1,124 @@
+            expect(result).toBe(null);
+        });
+        it('should return created user if user is deleted', async () => {
+            mockUserService.users = [{ ...testUser1, isDeleted: true }, { ...testUser2 }];
+            const result = await userController.createUser(testUserToCreate);
+            expect(result).toStrictEqual(testUser1);
+        });
+    });
+
+    describe('deleteUser', () => {
+        it('should return 1 if user exists', async () => {
+            mockUserService.users = [{ ...testUser1 }, { ...testUser2 }];
+            const result = await userController.deleteUser(mockUserId);
+            expect(result).toBe(1);
+        });
+        it('should return 0 if user does not exist', async () => {
+            mockUserService.users = [{ ...testUser1 }, { ...testUser2 }];
+            const result = await userController.deleteUser('wrongId');
+            expect(result).toBe(0);
+        });
+        it('should return 0 if user is already deleted', async () => {
+            mockUserService.users = [{ ...testUser1, isDeleted: true }];
+            const result = await userController.deleteUser(mockUserId);
+            expect(result).toBe(0);
+        });
+    });
+
+    describe('updateUser', () => {
+        it('should return 1 if user exists', async () => {
+            mockUserService.users = [{ ...testUser1 }, { ...testUser2 }];
+            const result = await userController.updateUser(mockUserId, testUserToUpdate);
+            expect(result).toBe(1);
+        });
+        it('should return 0 if user does not exist', async () => {
+            mockUserService.users = [{ ...testUser1 }, { ...testUser2 }];
+            const result = await userController.updateUser('wrongId', testUserToUpdate);
+            expect(result).toBe(0);
+        });
+        it('should return 0 if user is deleted', async () => {
+            mockUserService.users = [{ ...testUser1, isDeleted: true }];
+            const result = await userController.updateUser(mockUserId, testUserToUpdate);
+            expect(result).toBe(0);
+        });
+    });
+
+    describe('getAutoSuggestUsers', () => {
+        it('should return users with "user" substring and limit equals to 2', async () => {
+            mockUserService.users = [{ ...testUser1 }, { ...testUser2 }];
+            const result = await userController.getAutoSuggestUsers({ loginSubstring: 'user', limit: 2 });
+            expect(result).toStrictEqual([testUser1, testUser2]);
+        });
+        it('should return one user with "user" substring and limit equals to 1', async () => {
+            mockUserService.users = [{ ...testUser2 }, { ...testUser1 }];
+            const result = await userController.getAutoSuggestUsers({ loginSubstring: 'user', limit: 1 });
+            expect(result).toStrictEqual([testUser1]);
+        });
+        it('should return users with "user1" substring and limit equals to 2', async () => {
```

Comment:
Since the only thing `UserController` does is playing proxy role for `userService`, the unit tests for `UserController` should only check that `userService` used the right way (e.g. its methods are called with correct arguments).

These tests are rather testing the implementation of `MockUserService`